### ISSUE TITLE
Allow reminders in all channels

### ISSUE THREAD

**Issue by Numerlor** [python-discord/bot#1748]
Currently the reminder feature is restricted to #bot-commands for non-staff which makes it a bit less convenient to use quickly and be able to jump back to the conversation itself without copying the message link manually.

We can allow them everywhere with reminders being sent in #bot-commands instead of the channel the reminder was set in, to prevent unnecessary out of context messages from popping up (which already happens to some degree from remainders by staff)
An additional staff only channel argument would be accepted to control where the reminder is sent, that would also be required when used from staff channels

**Comment by NIRDERIi** [comment]
I see no reason why to enable it to every channel, I don't see how reminder, which is not a real Python-related command, should be enabled in Topical/any other channel other than bot commands, which is the most suited place for this command.

**Comment by janine9vn** [comment]
Yeah, I'd like to go with our "allow everywhere until proven problematic" policy. The only channel I might be iffy about enabling it is in python-general, but I can be convinced otherwise. 

I frequently see people trying to set themselves reminders to check in on an issue, or come back to some resource, or whatever else only to get told by the bot to go to bot-commands. I think having reminders available in non-py-gen channels will be a better experience for people.

I'm not even that sold of the disruptiveness of reminders being sent in the same channel, but I'm also fine if it gets routed through bot commands.

**Comment by HassanAbouelela** [comment]
The remind command is used quite functionally, at least within staff and contribution channels from my experience. If you check the git history for the latest change to that decorator, you’d see we were enabling it for open source channels.

I’d like to see it be global, though in slower channels (topical, open source, development, staff, etc), I think I don’t mind having it show up in channel.

**Comment by MarkKoz** [comment]
This is a good suggestion as it will really improve the usability of the command for the masses. A few points which are open to discussion:

1. In cases where it's not sending it to the same channel, is it feasible to first try to DM before resorting to sending it in `#bot-commands`? This might get complicated when multiple roles/users are mentioned in the reminder, but note that the feature is limited to Helpers+.
2. It should never send a reminder in a help channel, even if invoked by staff. The re-usable nature of these channels means the reminder may end up disrupting an unrelated help session.
3. If the reminder is sent in the same channel, it should reply (which is a WIP enhancement). Otherwise, it should include a jump link. In fact, it may be good to include the jump link in all cases for consistency's sake.

**Comment by TizzySaurus** [comment]
> This is a good suggestion as it will really improve the usability of the command for the masses. A few points which are open to discussion:
> 
> 1. In cases where it's not sending it to the same channel, is it feasible to first try to DM before resorting to sending it in `#bot-commands`? This might get complicated when multiple roles/users are mentioned in the reminder, but note that the feature is limited to Helpers+.
> 2. It should never send a reminder in a help channel, even if invoked by staff. The re-usable nature of these channels means the reminder may end up disrupting an unrelated help session.
> 3. If the reminder is sent in the same channel, it should reply (which is a WIP enhancement). Otherwise, it should include a jump link. In fact, it may be good to include the jump link in all cases for consistency's sake.

1. I think it's best to reply in #bot-commands rather than going through the effort of trying to DM. This also saves the issue of when the reminder has mentions.

2. Completely agree. We'd need to have some system to allow invocation in help-channels, but ensure that the reminder itself is sent in #bot-commands

3. I agree that it should both reply and have jump link. Main reason being that you can get things like "Failed to load message" (possibly more of a mobile issue) preventing jumping to the message just from the reply. Adding jump link means it's always possible.


In regards to allowing in #python-general, this can probably have the system as with help channels whereby it'll allow invocation but the reminder itself will be sent in #bot-commands, so as to not add any further "clutter" to the channel.

I'd be interested in implementing this once we have all the details ironed-out.
